src/plugins/vis_type_table/public/types.ts

```diff
@@ -48,7 +48,7 @@ export interface TableVisConfig extends TableVisParams {
 }
 
 export interface TableVisParams {
-  perPage: number;
+  perPage: number | '';
```

Comment:
this is a surprising type. Could we refactor to use `perPage?: number` instead?

Reply:
Yeah, this is because in table_vis_options.tsx, we use NumberInputOption component

```
<NumberInputOption
        label={
          <>
            <FormattedMessage
              id="visTypeTable.params.perPageLabel"
              defaultMessage="Max rows per page"
            />{' '}
            <EuiIconTip
              content="Leaving this field empty means it will use number of buckets from the response."
              position="right"
            />
          </>
        }
        isInvalid={!isPerPageValid}
        min={1}
        paramName="perPage"
        value={stateParams.perPage}
        setValue={setValue}
      />
```

which has the props defined as

```
interface NumberInputOptionProps<ParamName extends string> {
  disabled?: boolean;
  error?: ReactNode;
  isInvalid?: boolean;
  label?: React.ReactNode;
  max?: number;
  min?: number;
  paramName: ParamName;
  step?: number;
  value?: number | '';
  'data-test-subj'?: string;
  setValue: (paramName: ParamName, value: number | '') => void;
}

```
it requires `setValue` value type to be `number | ''`

Previously, it will show a type error
```
Type '<T extends keyof TableVisParams>(paramName: T, value: TableVisParams[T]) => void' is not assignable to type '(paramName: "perPage", value: number | "") => void'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'number | ""' is not assignable to type 'number'.
      Type 'string' is not assignable to type 'number'.ts(2322)
number_input.tsx(45, 3): The expected type comes from property 'setValue' which is declared here on type 'IntrinsicAttributes & NumberInputOptionProps<"perPage">'
```